### PR TITLE
[FIX] Resolve Timenode auto start race condition

### PR DESCRIPTION
**Pull request by adibas03**

### CODE REVIEW

**Comment by adibas03** [comment]
now tested, and ready

File: app/js/eac-worker.js [hunk 1]
```diff
@@ -51,12 +51,27 @@ class EacWorker {
     );
   }
 
-  startScanning() {
+  async awaitalarmClientInitialized () {
```
Comment (Kuzirashi) [inline app/js/eac-worker.js]: If you use camelCase then it should be:
`awaitAlarmClientInitialized`

File: app/js/eac-worker.js [hunk 2]
```diff
@@ -51,12 +51,27 @@ class EacWorker {
...
+    if (!this.alarmClient || !this.alarmClient.start || typeof this.alarmClient.start !== 'function') {
+      let Promises = new Promise((resolve) => {
```
Comment (Kuzirashi) [inline app/js/eac-worker.js]: There is no need for declaring variable `Promises`:
1. Name is misleading - there is only ONE promise - that's why in the first place it shouldn't be called `Promises`
2. After declaring variable the only one use is returning - just after declaration - it's better to return new promise without assigning it to variable first.

File: app/stores/KeenStore.js [hunk 3]
```diff
@@ -52,20 +52,32 @@ export class KeenStore {
     this.pollActiveTimeNodesCount();
   }
 
+  async awaitKeenInitialized () {
+    if (!this.networkId || !this.analysisClient || !this.trackingClient) {
+      let Promises = new Promise((resolve) => {
```
Comment (Kuzirashi) [inline app/stores/KeenStore.js]: There is no need for declaring variable Promises:

Name is misleading - there is only ONE promise - that's why in the first place it shouldn't be called Promises
After declaring variable the only one use is returning - just after declaration - it's better to return new promise without assigning it to variable first.

File: app/stores/TimeNodeStore.js [hunk 4]
```diff
@@ -125,9 +125,23 @@ export default class TimeNodeStore {
     this._keenStore.sendActiveTimeNodeEvent(this.getMyAddress(), this.getMyAttachedAddress());
   }
 
-  startScanning() {
+  async awaitScanReady() {
+    if (!this.eacWorker || this.eacWorker === null || !this._keenStore || this._keenStore === null ) {
+      let Promises = new Promise((resolve) => {
```
Comment (Kuzirashi) [inline app/stores/TimeNodeStore.js]: There is no need for declaring variable Promises:

Name is misleading - there is only ONE promise - that's why in the first place it shouldn't be called Promises
After declaring variable the only one use is returning - just after declaration - it's better to return new promise without assigning it to variable first.

**Pull request merged**